compiler/noirc_frontend/src/hir/resolution/import.rs

```diff
@@ -14,12 +15,17 @@ pub struct ImportDirective {
     pub is_prelude: bool,
 }
 
-pub type PathResolution = Result<PerNs, PathResolutionError>;
+pub type PathResolution =
+    Result<(ModuleId, PerNs, Option<PathResolutionError>), PathResolutionError>;
```

Comment:
```suggestion
pub enum PathResolution {
    Resolved {
      module_id: ModuleId,
      namespace: PerNs;
      errors: Vec<PathResolutionError>
    }
    Unresolved {
          errors: Vec<PathResolutionError>
    }
}
```

Thinking that we may want to refactor this into an enum to make it easier for us to pass back warnings/errors.